### PR TITLE
338 back function documentation

### PR DESCRIPTION
**Pull request by Renato549**
**PR Data:**
Task ID: (type the id of the development activity in the plan spreadsheet)

Problem: (description of what this PR is solving)

Solution: (description of how this PR solves the problem)

**PR Submitter Checklist:**

Submitter Name: (your name)

Before marking the PR as ready and asking for somehow to review it, make sure **ALL** steps
are checked. You are responsible for ensuring the answers to all questions are true.

If you make changes to the PR, you **HAVE** to make sure all steps are stil true.

- [x] You added the development minutes to the timelog in the activity plan ?
- [x] All required workflows pass correctly ?
- [ ] The branch/PR is in good state
  - [x] No "137 commits"
  - [x] No conflicts with `develop`
  - [x] If you need help, see: https://pastebin.com/tmbHpGDF
- [x] This PR completely marks as finished the related task ?
- [x] This PR does absolutely no work that isn't directly related to the related task ?
- [x] This PR conforms to the coding standards ?
  - [x] Frontend
    - [x] Components that concern about layout and style, are **100%** pure
- [x] This PR includes automated testing following these points ?
  - [x] Backend
    - [x] New functions that interact with the database have related tests in `__tests__/app/*`
    - [x] New endpoints have related tests in `__tests__/controller/*`
      - [x] Tests for authentication if needed
      - [x] Tests for validation if needed, either for body or search parameters
  - [x] Frontend
    - [x] Components are rendered using the `RenderTest` fixture at `fixtures.tsx` (see authentication.test.tsx for example)
    - [x] Test isn't just a render call
- [x] You manually tested any changes that cannot be tested in the automated tests ?
- [x] If this PR added new features that might be useful for other developers (for example, a new API to send s3 files)
  - [x] Did you add documentation on the root directory README.md ?
- [x] If this PR needs new data to show how it works on staging
  - [x] Did you add test entities in the seeds function at `seeds/index.ts` ?

**PR Reviewer Checklilst:**

Reviewer Name:

Before approving and merging the PR, make sure **ALL** steps are checked. You are
responsible for ensuring the answers to all questions are true.

If the submitter makes updates to the PR, you **MUST** go through all steps again.

- [ ] You added your name to this description in the "Reviewer name" field.
- [ ] You tested the changes locally
  - [ ] You checked out the PR locally
  - [ ] You ran the development server
  - [ ] You made sure everything starts up correctly
  - [ ] If the automated tests aren't enough, you manually confirmed the changes work as expected
  - [ ] You are sure all changes are either tested manually by you, or in automated tests
- [ ] You reviewd all changes and files manually
  - [ ] Code conforms to coding standards
- [ ] You are sure PR is ready to be merged to staging
- [ ] You are aware merging this PR will result in automatic changes to the staging deployment
- [ ] You marked your test and review minutes in the activity plan
- [ ] You approved and merged the PR

### CODE REVIEW

**Review by A13xMtzL — DISMISSED** [review]
Aplica, por favor:
`yarn lint:fix`
Ya que hay algunos problemas con el linter

File: ram-back/src/controller/group.ts [hunk 1]
```diff
@@ -25,12 +25,35 @@ import { authMiddleware } from "./user";
...
+`name`, `description`, and `imageUrl`, all of which are expected to be strings. This schema is later
+used as middleware to validate the request body before processing the update request. */
 const updateParameters = j.object({
   name: j.string(),
   description: j.string(),
   imageUrl: j.string(),
 });
 
+/**
+ * This is a middleware function in TypeScript that validates the request body using a schema and sends
```
Comment (monica-ayala) [inline ram-back/src/controller/group.ts:39]: los middlewares para obtener los datos correctos en el body con un schema no creo deberían de tener tanta documentación. Son 10 líneas y es código que se repite bastante.

File: ram-back/src/controller/group.ts [hunk 2]
```diff
@@ -104,6 +164,12 @@ groupsRouter.delete(
   },
 );
 
+/* The above code is defining a route for creating a new group. It requires authentication with the
```
Comment (monica-ayala) [inline ram-back/src/controller/group.ts:167]: Iguale estas pueden ser muy repetitivas

File: ram-back/src/controller/prospect.ts [hunk 3]
```diff
@@ -14,13 +14,35 @@ import { ProspectEnt } from "../entities/prospect.entity";
...
+also be strings. This schema is then used in the `prospectParametersMiddleware` middleware function
+to validate the request body before processing the request. */
 const prospectParameters = j.object({
   name: j.string().required(),
   firstSurname: j.string().required(),
   secondSurname: j.string().required(),
   comentary: j.string().optional(),
   statusId: j.string().optional(),
 });
+/**
```
Comment (monica-ayala) [inline ram-back/src/controller/prospect.ts:29]: Validaiton with schema middleware x2

File: ram-back/src/controller/user-delivery.ts [hunk 4]
```diff
@@ -60,6 +80,17 @@ userDeliveryRouter.get(
   },
 );
 
+/* This code defines a POST endpoint for the user delivery router that requires authentication using
+the `authMiddleware()` function with the `neededRoles` option set to `[UserRole.REGULAR]`. It takes
```
Comment (monica-ayala) [inline ram-back/src/controller/user-delivery.ts:84]: demasiada documentación

**Review by monica-ayala — CHANGES_REQUESTED** [review]
La herramienta a veces escribe demasiada documentación. Recomendaría que específicamente los middlewares para parámetros (los que hacen una validación con un schema y regresan BAD_DATA) sí se acorte la documentación. El resto puede quedarse así

**Review by monica-ayala — APPROVED** [review]
Algo extenso but lgtm

**Pull request merged**